### PR TITLE
Support janitor proxy in swan

### PR DESCRIPTION
**Pull request by liuzhengbaihe**

### CODE REVIEW

**Comment by upccup** [comment]
@liuzhengbaihe  ./vendor/github.com/Dataman-Cloud/swan-janitor/vendor/  this dir should be removed.   And add the janitor dependent in swan vendor dir.

File: main.go [hunk 1]
```diff
@@ -90,8 +91,8 @@ func main() {
 			Usage: "raft cluster peers addr",
 		},
 		cli.BoolFlag{
-			Name:  "enable-dns-proxy",
-			Usage: "enable dns proxy or not",
+			Name:  "enable-dns-janitor",
```
Comment (cmingxu) [inline main.go]: enable-proxy

File: main.go [hunk 2]
```diff
@@ -114,6 +115,10 @@ func main() {
 			Value: "sched",
 			Usage: "select engine,framework|sched",
 		},
+		cli.BoolFlag{
+			Name:  "enable-janitor-proxy",
```
Comment (cmingxu) [inline main.go]: enable-proxy would be fine

File: main.go [hunk 3]
```diff
@@ -123,6 +128,13 @@ func main() {
 
 		setupLogger(config.LogLevel)
 
+		// setup janitor service
+		if config.Janitor.EnableJanitorProxy {
```
Comment (cmingxu) [inline main.go]: put this into manager.go

File: src/config/config.go [hunk 4]
```diff
@@ -70,6 +73,12 @@ type Raft struct {
 	StorePath string `json:"store_path"`
 }
 
+type Janitor struct {
+	EnableJanitorProxy bool   `json:"enable-janitor-proxy"`
+	ListenerMode       string `json:"listener_mode"`
+	ListenerPort       string `json:"listener_port"`
```
Comment (cmingxu) [inline src/config/config.go]: `Port` would be fine

File: src/janitor/janitor.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+	jconfig "github.com/Dataman-Cloud/swan-janitor/src/config"
+	"github.com/Dataman-Cloud/swan-janitor/src/janitor"
+)
+
+type Janitor struct {
+	janitorConfig jconfig.Config
+	janitorServer *janitor.JanitorServer
+}
+
```
Comment (cmingxu) [inline src/janitor/janitor.go]: this file should not exists, as `janitor` is merely a third-party module.

**Comment by upccup** [comment]
👍

**Pull request merged**